File: 349_intersection_of_two_arrays.py
class Solution:
    def intersection(self, nums1: List[int], nums2: List[int]) -> List[int]:
        lesser = []
        greater = []
        solution = []
        if len(nums1) <= len(nums2):
            lesser = nums1
            greater = nums2
        else:
            lesser = nums2
            greater = nums1
        for i in range(len(lesser)):
            if lesser[i] not in solution:
                if lesser[i] in greater:
                    solution.append(lesser[i])
        return solution

File: 349_intersection_of_two_arrays.cpp
class Solution {
public:
    vector<int> intersection(vector<int>& nums1, vector<int>& nums2) {
        std::vector<int> lesser = {};
        std::vector<int> greater = {};
        std::vector<int> solution = {};
        if (nums1.size() <= nums2.size()) {
            lesser = nums1;
            greater = nums2;
        } else {
            lesser = nums2;
            greater = nums1;
        }
        for (int i = 0; i < lesser.size(); i++) {
            if (not std::ranges::contains(solution, lesser[i])) {
                if (std::ranges::contains(greater, lesser[i])) {
                    solution.push_back(lesser[i]);
                }
            }
        }
        return solution;
    }
};